subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -57,6 +57,19 @@ interface Identity {
         String getUniqueId();
     }
 
+    default IdentifyBuildOperationDetails identifyOperationDetails() {
+        return new IdentifyBuildOperationDetails() {
+            @Override
+            public Class<?> getWorkType() {
+                return UnitOfWork.this.getClass();
+            }
+        };
+    }
+
+    interface IdentifyBuildOperationDetails {
+        Class<?> getWorkType();
+    }
```

Comment:
❓ Is it necessary to add this complexity here? Can we use `visitImplementation()` instead?

Reply:
The point here is that different work types provide different sets of information in the details of the identify build operation. For example, artifact transforms will provide the from and to attributes. This is why the work itself needs to supply the identify operation details.

Is that the complexity you have been referring to? Or something else?

Reply:
As we discussed, let's emit a progress event in `AbstractTransformerExecution.identify()` for this purpose instead.